sdks/python/apache_beam/transforms/window.py

```diff
@@ -505,7 +527,8 @@ def get_window_coder(self):
     return coders.IntervalWindowCoder()
 
   def merge(self, merge_context):
-    to_merge = []
+    # type: (WindowFn.MergeContext) -> None
+    to_merge = []  # type: List[Union[IntervalWindow, GlobalWindow]]
```

Comment:
List[BoundedWindow]

Reply:
Changing this and the other use of `Union[IntervalWindow, GlobalWindow]` to `BoundedWindow` results in these errors:

```
apache_beam/transforms/window.py:541: error: "BoundedWindow" has no attribute "start"  [attr-defined]
apache_beam/transforms/window.py:543: error: "BoundedWindow" has no attribute "start"  [attr-defined]
apache_beam/transforms/window.py:550: error: "BoundedWindow" has no attribute "start"  [attr-defined]
apache_beam/transforms/window.py:557: error: "BoundedWindow" has no attribute "start"  [attr-defined]
```

The docs for `BoundedWindow` say that it's for timestamps in the range (-infinity, end).  

How about we add a `start` property to `BoundedWindow`:

```python
class BoundedWindow(object):
  """A window for timestamps in range (-infinity, end).

  Attributes:
    end: End of window.
  """

  def __init__(self, end):
    # type: (Union[int, float, Timestamp]) -> None
    self._end = Timestamp.of(end)

  @property
  def start(self):
    # type: () -> Timestamp
    return MAX_TIMESTAMP
```